tools/automation/coverage/run_command_coverage.py

```diff
@@ -0,0 +1,233 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from __future__ import print_function
+
+import collections
+import json
+import os
+import os.path
+import sys
+
+import automation.tests.nose_helper as automation_tests
+import automation.utilities.path as automation_path
+
+
+def init(root):
+    command_coverage = root.add_parser('command', help='Examine command and parameter test coverage.')
+    command_coverage.add_argument('--prefix', dest='prefix', default='', help='Filter results by string prefix.')
+    command_coverage.add_argument('--report', action='store_true', help='Display a report of summary statistics for the run.')
+    command_coverage.add_argument('--untested-params', help='Return commands for which any of these parameters (space-separated) are untested.')
+    command_coverage.add_argument('--projection', help='Filter output to only return the specific properties (space-separated).')
+    command_coverage.set_defaults(func=run_command_coverage)
+
+
+def _build_coverage_data(executed_commands):
+    coverage_data = {}
+    for command in executed_commands:
+        command_tokens = []
+        param_tokens = []
+        is_command = True
+        for token in command.split():
+            if token.startswith('-'):
+                param_tokens.append(token)
+                is_command = False
+            elif is_command:
+                command_tokens.append(token)
+            else:
+                # this would be a value token... meh.
```

Comment:
What does this mean?

Reply:
`COMMAND COMMAND -OPT VALUE --OPT VALUE VALUE VALUE --OPT`

The script only cares about finding all the command tokens and which option tokens were used. It doesn't check "value coverage" (i.e. that all options in an enum are tested).  So as it is parsing the tokens, if it finds one that is a value it simply ignores it for now.  If an argument could take three values, as long as at least one value is tested then the parameter is "tested" as far as this script is concerned.